addons/sale/static/src/js/sale_product_field.js

```diff
@@ -39,9 +39,10 @@ export class SaleOrderLineProductField extends Many2OneField {
         // product form should be accessible if the widget field is readonly
         // or if the line cannot be edited (e.g. locked SO)
         return (
-            this.props.readonlyField ||
-            (this.props.record.model.root.activeFields.order_line &&
-                this.props.record.model.root._isReadonly("order_line"))
+            this.props.context?.is_downpayment !== true && ( this.props.readonlyField || (
```

Comment:
Can't you access the field in the record data instead of relying on the context ?

Reply:
I can indeed. thx.